Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/webapp/deploymentslot/DeploymentSlotNode.java

```diff
@@ -32,24 +39,87 @@
     protected final String webAppId;
     protected final String slotName;
 
-
     public DeploymentSlotNode(final String slotId, final String webAppId, final DeploymentSlotModule parent,
-                              final String name, final String state, final String os,
-                              final String subscriptionId, final String hostName) {
-        super(slotId, name, LABEL, parent,subscriptionId, hostName, os, state);
+                              final String name, final String state, final String os, final String subscriptionId,
+                              final String hostName) {
+        super(slotId, name, LABEL, parent, subscriptionId, hostName, os, state);
         this.webAppId = webAppId;
         this.slotName = name;
         this.presenter = new DeploymentSlotNodePresenter();
         this.presenter.onAttachView(this);
     }
 
+    @Override
+    public List<NodeAction> getNodeActions() {
+        getNodeActionByName(ACTION_SWAP_WITH_PRODUCTION).setEnabled(this.state == WebAppBaseState.RUNNING);
+        return super.getNodeActions();
+    }
+
     @Override
     protected void loadActions() {
-        // todo
+        // todo: why only the stop action has icon?
+        addAction(ACTION_STOP, getIcon(this.os, this.label, WebAppBaseState.STOPPED),
+            createBackgroundActionListener("Stopping Deployment Slot", () -> stop()));
+        addAction(ACTION_START, createBackgroundActionListener("Starting Deployment Slot", () -> start()));
+        addAction(ACTION_RESTART,
+            createBackgroundActionListener("Restarting Deployment Slot", () -> restart()));
+        addAction(ACTION_SWAP_WITH_PRODUCTION,
+            createBackgroundActionListener("Swapping with Production", () -> swapWithProduction()));
+        addAction(ACTION_OPEN_IN_BROWSER, new NodeActionListener() {
+            @Override
+            protected void actionPerformed(NodeActionEvent e) {
+                DefaultLoader.getUIHelper().openInBrowser("http://" + hostName);
+            }
+        });
+
+        super.loadActions();
+    }
+
+    @Override
+    protected void onNodeClick(NodeActionEvent e) {
+        // RefreshableNode refresh itself whenever the node is clicked.
+        // The deployment slot node is just a single node for the time being.
+        // Override the function to do noting to disable the auto refresh functionality.
+    }
+
+    private void start() {
+        try {
+            presenter.onStartDeploymentSlot(this.subscriptionId, this.webAppId, this.slotName);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
What about the error handling? I don't thin doing nothing is a correct way, because if exception occurs, the app is not started. The user should be at least notified somehow.

Reply:
I will add a todo item here and let's discuss the error handling for both the deployment node operations and web app node operations.